src/app/templates/client/Client.jsx

```diff
@@ -38,27 +40,80 @@ function SystemEmojiFeature() {
 function Client() {
   const [isLoading, changeLoading] = useState(true);
   const [loadingMsg, setLoadingMsg] = useState('Heating up');
+  const [isTouchingSide, setTouchingSide] = useState(false);
+  const [sideMoved, setSideMoved] = useState(0);
   const classNameHidden = 'client__item-hidden';
+  const classNameSided = 'client__item-sided';
 
   const navWrapperRef = useRef(null);
   const roomWrapperRef = useRef(null);
 
   function onRoomSelected() {
-    navWrapperRef.current?.classList.add(classNameHidden);
     roomWrapperRef.current?.classList.remove(classNameHidden);
+    navWrapperRef.current?.classList.add(classNameSided);
   }
   function onNavigationSelected() {
-    navWrapperRef.current?.classList.remove(classNameHidden);
-    roomWrapperRef.current?.classList.add(classNameHidden);
+    navWrapperRef.current?.classList.remove(classNameSided);
+    setTimeout(() => roomWrapperRef.current?.classList.add(classNameHidden), 250);
+  }
+  // Touch handlers for window object. If the touch starts at 10% of the left of the screen, it will trigger the swipe-right-menu.
+  let lastTouch = 0, sideVelocity = 0;
```

Comment:
Should this be defined in the Client component or stored in a ref to persist across re-renders? Or is it intentionally designed to reset on every render?

Reply:
Now using `useState` for both https://github.com/cinnyapp/cinny/pull/1647/commits/677ad454ef2f434b6cd08958be1e6930182657a3